readme.md

```diff
@@ -35,31 +35,104 @@ Installation
 ------------
 The framework can be included into your project with composer. Run `composer require uhi67/umvc:dev-master`
 
-First steps
------------
-1. Create `composer.json` of your application, and include uh67/umvc, e.g `composer init --name myname/myapp --require uhi67/umvc:*`
-2. run `composer update`
+First steps to build your application using UMVC
+------------------------------------------------
+
+**Warning: This part is under construction.**
+
+1. Create `composer.json` of your application, and include `uh67/umvc`, e.g `composer init --name myname/myapp --require uhi67/umvc:*`
+2. Run `composer update`
+3. Copy `vendor/uh67/umvc/app` to your application's root. This is the launcher of the CLI commands. 
+4. Copy `vendor/uh67/umvc/www/index.php` and `.htaccess` to your application's www directory. This is the router of the web interface. 
+5. Create your application's config in the `config/config.php` file, see template in `vendor/uh67/umvc/config/config-template.php`
+6. Create a `runtime` directory writable by the webserver to place temporary files.
+7. Create the `www/assets` directory writable by the webserver to place cached asset files of various components.
+
+#### Build your application using MVC pattern
+
+1. Create your controllers in the `controllers` dir, using `\app\controllers` namespace, and derive them from `uhi67\umvc\Controller`
+2. Create the views in the `views` dir, in simple PHTML format, and organize them `views/controller/action.php` structure.
+3. Create your models in the `models` directory. Database models are `\uhi67\umvc\Model`, database-less models are `\uhi67\umvc\BaseModel`
+                                         
+#### Other basic principles
+
+1. Migrations are for keeping database changes in sync with your source code in the VC. Place migration steps into `migrations` directory.
```

Comment:
Migrations are for versioning and recording database changes in source code. [...]